modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/flamingos2/Flamingos2.scala

```diff
@@ -34,13 +35,15 @@ final case class Flamingos2(f2Controller: Flamingos2Controller) extends Instrume
 
   // FLAMINGOS-2 does not support abort or stop.
   override def observe(config: Config): SeqObserve[ImageFileId, ObserveCommand.Result] = Reader {
-    fileId => f2Controller.observe(fileId).map(_ => ObserveCommand.Success)
+    fileId => f2Controller.observe(fileId, calcObserveTimeout(config)).map(_ => ObserveCommand.Success)
   }
 
   override def configure(config: Config): SeqAction[ConfigResult] =
     fromSequenceConfig(config).flatMap(f2Controller.applyConfig).map(_ => ConfigResult(this))
 
   override def notifyObserveEnd: SeqAction[Unit] = f2Controller.endObserve
+
+  override def calcObserveTimeout(config: Config) = config.extract(OBSERVE_KEY / EXPOSURE_TIME_PROP).as[java.lang.Double].map(x => Seconds(x.toDouble)).getOrElse(Seconds(360))
```

Comment:
I thought `EXPOSURE_TIME_PROP` would be the requested exposure time. Does this mean the timeout is the same as the requested exposure time?

Reply:
No. I then add a readout timeout. But now I see I should have added that value here to make it consistent with the function name.

Reply:
In second thought, what I should change is the name of the function, because the time added to calculate the timeout depends on the underline Controller.